config/management/genesis/src/validator_config.rs

```diff
@@ -1,123 +1,60 @@
         )?;
+        let owner_key = owner_shared_storage.ed25519_key(OWNER_KEY)?;
+        let owner_account = account_address::from_public_key(&owner_key);
 
-        // Fetch keys from storage
-        let (consensus_key, fullnode_network_key, validator_network_key) =
-            fetch_keys_from_storage(storage_name, &self.validator_backend.validator_backend)?;
-
-        // Append ln-noise-ik and ln-handshake protocols to base network addresses
-        // and encrypt the validator address.
-        let validator_address = self
-            .validator_address
-            .append_prod_protos(validator_network_key, HANDSHAKE_VERSION);
-        let raw_enc_validator_address = encode_validator_address(
-            validator_address,
-            &owner_account,
-            sequence_number,
-            &TEST_SHARED_VAL_NETADDR_KEY,
-            TEST_SHARED_VAL_NETADDR_KEY_VERSION,
-            idx_addr,
-        )?;
-
-        let fullnode_address = self
-            .fullnode_address
-            .append_prod_protos(fullnode_network_key, HANDSHAKE_VERSION);
-        let raw_fullnode_address = encode_address(fullnode_address)?;
-
-        let validator_config = libra_types::validator_config::ValidatorConfig {
-            consensus_public_key: consensus_key,
-            validator_network_identity_public_key: validator_network_key,
-            validator_network_address: raw_enc_validator_address,
-            full_node_network_identity_public_key: fullnode_network_key,
-            full_node_network_address: raw_fullnode_address,
-        };
-
-        // Create the validator config script for the validator node
-        let validator_config_script =
-            create_validator_config_script(owner_account, validator_config);
-
-        // Create and sign the validator-config transaction
-        let validator_config_tx = create_transaction(
-            storage_name,
-            &self.validator_backend.validator_backend,
-            sequence_number,
-            self.chain_id,
-            "validator-config",
-            validator_config_script,
+        let mut validator_storage = StorageWrapper::new(
+            self.validator_config.validator_backend.name(),
+            &self.validator_config.validator_backend.validator_backend,
         )?;
-        let validator_config_tx = Transaction::UserTransaction(validator_config_tx);
+        validator_storage.set(OWNER_ACCOUNT, Value::String(owner_account.to_string()))?;
 
-        // Write validator config to local storage to save for verification later on
-        let mut validator_storage =
-            StorageWrapper::new(storage_name, &self.validator_backend.validator_backend)?;
-        validator_storage.set(
-            constants::VALIDATOR_CONFIG,
-            Value::Transaction(validator_config_tx.clone()),
+        let txn = self.validator_config.build_transaction(
+            0,
+            self.fullnode_address,
+            self.validator_address,
+            false,
```

Comment:
This is sometimes contentious in languages, but I really wish that you could put here `reconfiguration=false`.  But, I don't think rust supports that

Reply:
no, no it doesn't, if you are really anal, you can make enums :)